bitcoin/src/blockdata/script.rs

```diff
@@ -1311,22 +1342,87 @@ impl ScriptBuf {
     #[inline]
     pub fn into_boxed_script(self) -> Box<Script> {
         // Copied from PathBuf::into_boxed_path
-        let rw = Box::into_raw(self.0.into_boxed_slice()) as *mut Script;
+        let rw = Box::into_raw(self.1.into_boxed_slice()) as *mut Script;
         unsafe { Box::from_raw(rw) }
     }
 }
 
+/// A [`ScriptBuf`] used as the script code when spending segwit utxos.
+///
+/// Please note, the script code created here does not contain the length byte, this is pre-pended
+/// during serialization by `consensus_encode` and in the sighash cache we use `consensus_encode` to
+/// encode the script codes created below - therefore the length byte is not present in the raw
+/// script returned here. To get a hex string use `crate::consensus::serialize_hex(&script_code)`.
```

Comment:
This seems to be stating the obvious - we don't store the length in any script, but OK.

Reply:
I spent a lot of hours in the last three days working out exactly why bytes were missing in places and added in others, perhaps this is common knowledge for others hacking on this codebase longer than me. This, an other similar comments where my effort to save the next guy some time. I can remove or trim them down if I've been too laborious though.

Reply:
Maybe we should have some general information about thins in the module or so. "None of the script types stores its size inline, to output the script with the size prepended as compact size call `consensus_encode()`."